ports/esp8266/main.c

```diff
@@ -49,6 +49,64 @@
 
 static char heap[38 * 1024];
 
+#if MICROPY_PY_EXCEPTION_REPORT
```

Comment:
Config names of the form `MICROPY_PY_xxx` are intended to be for things in the Python scope, like modules, types, etc.

Also, "exception" could be confused with Python exceptions.

I suggest a name like `MICROPY_HW_HARD_FAULT_DEBUG`

Reply:
Thanks!  Maybe the `MICROPY_PY_` / `MICROPY_HW_` notation details should be added to `CODECONVENTIONS.md` or to another document under `docs/develop/`.

I've noticed a few (I guess now historical) divergences though, which maybe could be normalised for v2?  E.g. `MICROPY_REPL_EMACS_KEYS` could really be named `MICROPY_PY_REPL_EMACS_KEYS` since it is entirely in the Python scope.